rio_tiler/io/image.py

```diff
@@ -0,0 +1,562 @@
+"""Image Reader."""
+
+import contextlib
+import math
+import warnings
+from typing import Any, Callable, Dict, List, Optional, Sequence, Union
+
+import attr
+import morecantile
+import numpy
+import rasterio
+from morecantile.utils import _parse_tile_arg
+from rasterio import windows
+from rasterio.crs import CRS
+from rasterio.enums import ColorInterp, Resampling
+from rasterio.io import DatasetReader, DatasetWriter, MemoryFile
+from rasterio.rio.overview import get_maximum_overview_level
+
+from ..errors import (
+    AlphaBandWarning,
+    ExpressionMixingWarning,
+    IncorrectTileBuffer,
+    NoOverviewWarning,
+)
+from ..expression import apply_expression, parse_expression
+from ..models import BandStatistics, ImageData, Info
+from ..types import BBox, DataMaskType, Indexes, NumType
+from ..utils import (
+    get_array_statistics,
+    get_bands_names,
+    has_alpha_band,
+    has_mask_band,
+    non_alpha_indexes,
+)
+from .base import BaseReader
+
+
+def imread(
+    src_dst: Union[DatasetReader, DatasetWriter],
+    height: Optional[int] = None,
+    width: Optional[int] = None,
+    indexes: Optional[Indexes] = None,
+    window: Optional[windows.Window] = None,
+    force_binary_mask: bool = True,
+    unscale: bool = False,
+    resampling_method: Resampling = "nearest",
+    post_process: Optional[
+        Callable[[numpy.ndarray, numpy.ndarray], DataMaskType]
+    ] = None,
+    nodata=None,
+) -> DataMaskType:
+    """Low level read function.
+
+    Like rio_tiler.reader.read but without WarpedVRT
```

Comment:
👆 

Note: `nodata` is not used right now but keep is there to keep consistency with `rio_tiler.reader.read`